monitor/sortedOracles.go

```diff
@@ -15,36 +16,55 @@ import (
 	"github.com/celo-org/kliento/contracts/helpers"
 )
 
+type stableTokenInfo struct {
+	address     common.Address
+	exchange    *contracts.Exchange
+	stableToken celotokens.CeloToken
+}
+
 type sortedOraclesProcessor struct {
-	ctx           context.Context
-	logger        log.Logger
-	sortedOracles *contracts.SortedOracles
-	exchange      *contracts.Exchange
+	ctx    context.Context
+	logger log.Logger
+
+	sortedOracles    *contracts.SortedOracles
+	stableTokenInfos map[celotokens.CeloToken]stableTokenInfo
 }
 
-func NewSortedOraclesProcessor(ctx context.Context, logger log.Logger, sortedOracles *contracts.SortedOracles, exchange *contracts.Exchange) *sortedOraclesProcessor {
-	return &sortedOraclesProcessor{
-		ctx:           ctx,
-		logger:        logger.New("contract", "SortedOracles"),
-		sortedOracles: sortedOracles,
-		exchange:      exchange,
+func NewSortedOraclesProcessor(ctx context.Context, logger log.Logger, sortedOracles *contracts.SortedOracles, exchanges map[celotokens.CeloToken]*contracts.Exchange, stableTokenAddresses map[celotokens.CeloToken]common.Address) (*sortedOraclesProcessor, error) {
+	stableTokenInfos := make(map[celotokens.CeloToken]stableTokenInfo)
+	for stableToken, stableTokenAddress := range stableTokenAddresses {
+		exchange, ok := exchanges[stableToken]
+		if !ok || exchange == nil {
```

Comment:
this should never happen right?

Reply:
Should never happen based off the behavior in kliento & `monitor/main.go` but just to be safe in the future I included this. Adding a comment to indicate that